library/src/main/java/com/google/android/material/motion/runtime/TargetScope.java

```diff
@@ -70,6 +72,22 @@ void commitPlan(PlanInfo plan) {
     performer.addPlan(plan.plan);
   }
 
+  void commitAddNamedPlan(NamedPlan plan, String name, Object target) {
+    // remove first
+    commitRemoveNamedPlan(name);
+    // then add
+    NamedPlanPerformance namedPerformer = namedCache.get(name);
+    if (namedPerformer == null) {
+      namedPerformer = (NamedPlanPerformance)createPerformer(plan, target);
+    }
```

Comment:
this is fine for now - can you add a `TODO: refactor getPerformer() and use it here` for me